### DISCUSSION TITLE
[Feature]: Viewing of Android Motion Photos

### DISCUSSION THREAD

**Discussion by eightys3v3n** [immich-app/immich#1677] [General]
### Feature detail
The ability to play HEIC motion photos from Apple and Android phones.
A play button somewhere that doesn't block the photo so auto-play can be toggled.

### Platform

Mobile App

**Comment by alextran1502** [comment]
Duplicated with #160

**Comment by eightys3v3n** [comment]
Ah, thank you. I missed the other one originally but just found it. Sorry for the duplicate.

**Comment by jakejoh** [comment]
I think this issue should remain open (maybe with a more precise title/description?) as #160 covers live photos from iOS devices, not Android's similar Motion Photos.

https://github.com/immich-app/immich/issues/160#issuecomment-1140525363

**Comment by alextran1502** [comment]
@jakejoh Agree, however, we don't have the mechanism to distinguish Motion Photos on Android from a normal photos.

**Comment by jakejoh** [comment]
at least Google Camera photos contain a xmp section like this:
```xml
<x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="Adobe XMP Core 5.1.0-jc003">
  <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <rdf:Description rdf:about=""
        xmlns:GCamera="http://ns.google.com/photos/1.0/camera/"
        xmlns:Container="http://ns.google.com/photos/1.0/container/"
        xmlns:Item="http://ns.google.com/photos/1.0/container/item/"
        xmlns:xmpNote="http://ns.adobe.com/xmp/note/"
      GCamera:MotionPhoto="1"
      GCamera:MotionPhotoVersion="1"
      GCamera:MotionPhotoPresentationTimestampUs="467400"
      xmpNote:HasExtendedXMP="5748706CE8036406D65595EBF4C31711">
      <Container:Directory>
        <rdf:Seq>
          <rdf:li rdf:parseType="Resource">
            <Container:Item
              Item:Mime="image/jpeg"
              Item:Semantic="Primary"
              Item:Length="0"
              Item:Padding="0"/>
          </rdf:li>
          <rdf:li rdf:parseType="Resource">
            <Container:Item
              Item:Mime="video/mp4"
              Item:Semantic="MotionPhoto"
              Item:Length="3589453"
              Item:Padding="0"/>
          </rdf:li>
        </rdf:Seq>
      </Container:Directory>
    </rdf:Description>
  </rdf:RDF>
</x:xmpmeta>
```
and are named ...MP.jpg
e.g. this one: [PXL_20221121_171540706.MP.jpg](https://user-images.githubusercontent.com/25611892/203121462-d9ea2719-27c4-47a8-b2c2-7e0ea5f2d424.jpg)

[Aves has some ways decting motion photos](https://github.com/deckerst/aves/blob/3f94a4f5a0835d3e721a6340b9d89d1e0f49af93/android/app/src/main/kotlin/deckers/thibault/aves/metadata/XMP.kt#L177)

**Comment by wjbeckett** [comment]
Absolutely need this for Android Motion photos.
Looks like there is a way to discern motion photos from normal photos.

I assume this would make it able to not have them double up in the timeline as well and "stack" the photos? That way when you view a motion photo in immich you could see the motion photo by default but also see the static photo as well to stop the duplication in the timeline.

**Comment by mammo0** [comment]
Hey,

if I'm right, the current state is that this feature is implemented for iOS.
I saw in the API docs that the file upload function contains a field for live / motion photo data (`livePhotoData`): https://documentation.immich.app/docs/api/upload-file

So, is the server capable of handling motion photo data (which is basically a video)? This would mean that the handling and identification of motion photos is done in the mobile App. Is this assumption correct?

Maybe I have some time in the next week to create a PR for this feature. At least for Google Camera photos, because I have a Pixel device. But before I have two questions:

1. Can someone of the maintainers give me a hint, where it's the best place in the code to implement this feature? I haven't worked with Dart yet, so some starting help would be great.
2. What kind of binary data expects should be in the above mentioned `livePhotoData` field? I guess video binary data. But which codec?

**Comment by JetSetDoritos** [comment]
Looks like this service open sourced how they discern Motion Photos

https://github.com/ente-io/motion_photos

**Comment by jrasm91** [comment]
Implemented a few releases back.
